src/main/java/org/opensrp/repository/postgres/LocationTagRepositoryImpl.java

```diff
@@ -0,0 +1,228 @@
+		
+		if (getPgLocationTag == null) {
+			org.opensrp.domain.postgres.LocationTag pgLocationTag = convert(locationTag);
+			pgLocationTag.setId(id);
+			locationTagMapper.updateByPrimaryKey(pgLocationTag);
+		} else {
+			throw new DuplicateKeyException("Location tag name already exists");
+			
+		}
+		
+	}
+	
+	@Override
+	public List<LocationTag> getAll() {
+		LocationTagExample locationTagExample = new LocationTagExample();
+		locationTagExample.createCriteria().andActiveIsNotNull();
+		List<org.opensrp.domain.postgres.LocationTag> pgLocationTagList = locationTagMapper.selectMany(locationTagExample,
+		    0, DEFAULT_FETCH_SIZE);
+		return convert(pgLocationTagList);
+	}
+	
+	@Override
+	public void safeRemove(LocationTag locationTag) {
+		if (locationTag == null) {
+			return;
+		}
+		
+		Long id = (Long) retrievePrimaryKey(locationTag);
+		if (id == null) {
+			return;
+		}
+		
+		org.opensrp.domain.postgres.LocationTag pgLocationTag = convert(locationTag);
+		pgLocationTag.setId(id);
+		pgLocationTag.setActive(false);
+		locationTagMapper.updateByPrimaryKey(pgLocationTag);
+		
+	}
+	
+	@Override
+	public LocationTag getLocationTagByPrimaryKey(Long id) {
+		if (id == null) {
+			return null;
+		}
+		LocationTagExample example = new LocationTagExample();
+		example.createCriteria().andIdEqualTo(id);
+		List<org.opensrp.domain.postgres.LocationTag> locationTags = locationTagMapper.selectByExample(example);
+		return locationTags.isEmpty() ? null : convert(locationTags.get(0));
+	}
+	
+	@Override
+	public void safeRemove(Long id) {
+		LocationTag locationTag = getLocationTagByPrimaryKey(id);
+		org.opensrp.domain.postgres.LocationTag pgLocationTag = convert(locationTag);
+		if (pgLocationTag == null) {
+			return;
+		}
+		pgLocationTag.setId(id);
+		pgLocationTag.setActive(false);
+		locationTagMapper.updateByPrimaryKey(pgLocationTag);
+		
+	}
+	
+	@Override
+	public org.opensrp.domain.postgres.LocationTag getLocationTagByName(String name) {
+		if (StringUtils.isBlank(name)) {
+			return null;
+		}
+		
+		LocationTagExample locationTagExample = new LocationTagExample();
+		locationTagExample.createCriteria().andNameEqualTo(name);
+		List<org.opensrp.domain.postgres.LocationTag> locationTagList = locationTagMapper
+		        .selectByExample(locationTagExample);
+		
+		return isEmptyList(locationTagList) ? null : locationTagList.get(0);
+	}
+	
+	@Override
+	protected Object retrievePrimaryKey(LocationTag locationTag) {
+		Object uniqueId = getUniqueField(locationTag);
+		if (uniqueId == null) {
+			return null;
+		}
+		
+		String name = uniqueId.toString();
+		org.opensrp.domain.postgres.LocationTag pgLocationTag = getLocationTagByName(name);
+		
+		return pgLocationTag == null ? null : pgLocationTag.getId();
+	}
+	
+	@Override
+	protected Object getUniqueField(LocationTag locationTag) {
+		return locationTag == null ? null : locationTag.getName();
+	}
+	
+	@Override
+	public LocationTag get(String id) {
+		// TODO Auto-generated method stub
```

Comment:
Implement this or throw not implemented exception

Reply:
throw not implemented exception at this method